tvbwidgets/exporters/model_exporters.py

```diff
@@ -0,0 +1,200 @@
+            f.write(json.dumps(config_dict))
+
+    def _read_existing_config(self):
+        # type: () -> dict
+        """
+        reads a json file and returns the dict representing the json data or an empty dict
+        if the config file or data in it doesn't exist
+        """
+        file_name = self.file_name
+        config = {}
+        try:
+            with open(file_name, 'r+') as config_file:
+                config = json.loads(config_file.read())
+        except (FileNotFoundError, json.JSONDecodeError):
+            pass
+        return config
+
+    def _prepare_export_values(self):
+        # type: () -> dict
+        """
+        returns a json safe dict containing the model params from the exporter keys
+        """
+        values = {}
+        model_dict = self.model_instance.__dict__
+
+        for key in self.keys:
+            if model_dict.get(key, False):
+                values[key] = self.sanitize_property(model_dict[key])
+
+        values['model'] = self.model_instance.__class__.__name__
+
+        return values
+
+
+class PythonCodeExporter(ABCModelExporter):
+    file_name = 'model_instances.py'
+    numpy_import = 'import numpy'
+    models_import = 'from tvb.simulator.models import *'
+    instance_var_name = 'model_instance'
+
+    def __init__(self, model_instance, keys):
+        # type: (Model, list[str]) -> None
+        super(PythonCodeExporter, self).__init__(model_instance, keys)
+
+    @property
+    def default_config_name(self):
+        return 'default config'
+
+    def do_export(self):
+        class_name = self.model_instance.__class__.__name__
+        values = f'# {self.config_name}\n'
+        # assume that if the file exists the imports also exist
+        if not os.path.exists(self.file_name):
+            values += f'{self.numpy_import}\n{self.models_import}\n'
+
+        model_params = self.get_model_params()
+        values += f'{self.instance_var_name} = {class_name}({model_params})\n\n'
+
+        # open the file to append to existing saved model instances
+        with open(self.file_name, 'a') as f:
+            f.write(values)
+
+    def get_model_params(self):
+        model_params = ''
+        model_instance_dict = self.model_instance.__dict__
+
+        for key in self.keys:
+            if model_instance_dict.get(key, False):
+                model_params += f'{key}=numpy.array({model_instance_dict[key]}),'
+        # strip the last comma on params before returning
+        return model_params.rstrip(model_params[-1])
+
+
+def model_exporter_factory(exporter_type, model, keys):
+    # type: (str, Model, list[str]) -> ABCModelExporter
+    """
+    Factory for model exporter creation
+    """
+    exporter = MODEL_CONFIGURATION_EXPORTS.get(exporter_type, False)
+
+    if not exporter:
+        raise ModelExporterNotFoundError('Could not find any exporter of the selected type!')
+
+    if exporter == MODEL_CONFIGURATION_EXPORTS['Python script']:
```

Comment:
I thought MODEL_CONFIGURATION_EXPORTS was an Enum, I see now that it is a dict, so we do not really get rid of string hardcoding here.